snippets/java/charFreqCount.md

```diff
@@ -0,0 +1,85 @@
+# Count frequency of unique characters in a String
+
+## Code
+```Java
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.InputStreamReader;
+
+public class Main {
+    public static void main(String[] args) throws Exception {
+        Scanner scn = new Scanner(new InputStreamReader(System.in));
+        
+        //  string input
+        System.out.println("Enter String : ");
+        char[] str = scn.nextLine().toCharArray();
+        
+        //  Frequency map of string characters
+        int[] chars = new int[127 - 32];      //  readble ASCII characters are from 32 to 126 number
+        for (char c : str) {
+            chars[c - ' ']++;       //  ASCII(32) = ' '
+        }
+
+        //  output to write in file
+        StringBuilder res = new StringBuilder("Frequency of characters in input string is : \n\n");
+        for (int i = 0, N = chars.length; i < N; i++) {
+            if (chars[i] != 0) {    //  if frequency is not 0 then append in result string
+                res.append("character '" + (char)(' ' + i) + "' present :" + chars[i] + " times\n");
```

Comment:
I think a proper sentence will look better along with the corresponding changes in the output block. :)
```suggestion
                res.append("The character '" + (char)(' ' + i) + "' occurred " + chars[i] + " times.\n");
```